### PR TITLE
collect tests that dont mod 4

### PR DESCRIPTION
**Pull request by mmaxim**
cc @zapu

### CODE REVIEW

**Comment by mmaxim** [comment]
cc @oconnor663

**Comment by maxtaco** [comment]
i'm  not sure what will happen if we roll back the UPAK version

**Comment by maxtaco** [comment]
very frustrated about this, i don't know what's broken about my pr

out for most of the day thogh

File: Jenkinsfile [hunk 1]
```diff
@@ -408,6 +408,9 @@ def testGo(prefix) {
                 }
             }
         }
+        if (tests.size() > 0) {
```
Comment (jzila) [inline Jenkinsfile]: nice catch

**Comment by mmaxim** [comment]
@maxtaco LMKWYT

File: go/uidmap/uidmap_test.go [hunk 2]
```diff
@@ -177,14 +177,15 @@ func TestRanOutOfTime(t *testing.T) {
...
 	// server
+	t.Logf("tKB: %s", tKB)
 	uids = []keybase1.UID{tKB}
 	hit = false
 	results, err = uidMap.MapUIDsToUsernamePackages(context.TODO(), tc.G, uids, 0, 0, true)
 	require.NoError(t, err)
 	require.True(t, hit)
 	require.Equal(t, results[0].NormalizedUsername, libkb.NewNormalizedUsername("t_kb"))
 	require.Equal(t, results[0].FullName.FullName, keybase1.FullName("Joe Keybaser"))
-	require.Equal(t, results[0].FullName.EldestSeqno, keybase1.Seqno(1))
```
Comment (mmaxim) [inline go/uidmap/uidmap_test.go]: `uidmap` has also not been run in Jenkins, and this test is just broken I think. These users have an eldest of 0 not 1.

File: go/teams/service_helper.go [hunk 3]
```diff
@@ -139,7 +139,7 @@ func userVersionsToDetails(ctx context.Context, g *libkb.GlobalContext, uvs []ke
 	for i, uv := range uvs {
 		pkg := packages[i]
 		active := true
-		if pkg.FullName != nil && pkg.FullName.EldestSeqno > uv.EldestSeqno {
+		if pkg.FullName != nil && pkg.FullName.EldestSeqno != uv.EldestSeqno {
```
Comment (maxtaco) [inline go/teams/service_helper.go]: i think if `pkg.FullName.EldestSeqno == 0`, we want active to be `false`, no?  Should never have a reset user in the team.

File: go/uidmap/uidmap.go [hunk 4]
```diff
@@ -285,27 +292,33 @@ func (u *UIDMap) MapUIDsToUsernamePackages(ctx context.Context, g libkb.UIDMappe
...
-		// There are 3 important cases when we should go to network:
-		//
-		//  1. No username is found (up == nil)
-		//  2. No FullName found and we've asked to force network lookups (status == notFound && forceNetworkForNullNames)
-		//  3. The FullName found was stale (status == stale).
-		//
-		// Thus, if you provide forceNetworkForFullName=false, and fullNameFreshness=0, you can avoid
-		// the network trip as long as all of your username lookups hit the cache or are hardcoded.
-		if up == nil || (status == notFound && forceNetworkForFullNames) || (status == stale) {
+	if u.testNoCachingMode {
```
Comment (maxtaco) [inline go/uidmap/uidmap.go]: i'd say nuke this `if` arm... and...

File: go/uidmap/uidmap.go [hunk 5]
```diff
@@ -285,27 +292,33 @@ func (u *UIDMap) MapUIDsToUsernamePackages(ctx context.Context, g libkb.UIDMappe
...
+			}
+
+			// There are 3 important cases when we should go to network:
+			//
+			//  1. No username is found (up == nil)
+			//  2. No FullName found and we've asked to force network lookups (status == notFound && forceNetworkForNullNames)
+			//  3. The FullName found was stale (status == stale).
+			//
+			// Thus, if you provide forceNetworkForFullName=false, and fullNameFreshness=0, you can avoid
+			// the network trip as long as all of your username lookups hit the cache or are hardcoded.
```
Comment (maxtaco) [inline go/uidmap/uidmap.go]: `|| testNoCachingMode` here?

Comment (mmaxim) [reply go/uidmap/uidmap.go]: Keep in mind I was also racing again multiple invokations of `service.TeamGet`, where I needed to make sure to increment the time after every try. That wasn't looking so easy, so I decided to take the blunt approach with the caches here.

File: go/teams/service_helper.go (hunk 3 above)
Comment (mmaxim) [reply go/teams/service_helper.go]: I maintain that is how this works, but I can change it if you want. We don't go into the `team_members` table rewrite all memberships rows to 0 when someone resets.

Comment (maxtaco) [reply go/teams/service_helper.go]: ok, cool

File: go/uidmap/uidmap.go (hunk 5 above)
Comment (mmaxim) [reply go/uidmap/uidmap.go]: `pollForMembershipUpdate` this guy just spam runs `service.TeamGet`

Comment (maxtaco) [reply go/uidmap/uidmap.go]: i think my approach is just as blunt?

Comment (mmaxim) [reply go/uidmap/uidmap.go]: Well you also need to change line 305? Maybe less clean? Your call.

Comment (maxtaco) [reply go/uidmap/uidmap.go]: no, it'll get trampled over

Comment (mmaxim) [reply go/uidmap/uidmap.go]:
```
	// Always write these results out if the cached value is unset.
				// Or, see below for other case...
				writeResults := res[apiLookupIndex[i]].NormalizedUsername.IsNil()
```

Comment (mmaxim) [reply go/uidmap/uidmap.go]:
```
if writeResults {
					// Overwrite the row with whatever was returned from the server.
					res[apiLookupIndex[i]] = row
				}
```

**Pull request merged**